src/Sylius/Bundle/CoreBundle/spec/Sitemap/Service/SitemapBuilderSpec.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+ 
+namespace spec\Sylius\Bundle\CoreBundle\Sitemap\Service;
+ 
+use Doctrine\Common\Collections\ArrayCollection;
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+use Sylius\Bundle\CoreBundle\Sitemap\Factory\SitemapFactoryInterface;
+use Sylius\Bundle\CoreBundle\Sitemap\Model\SitemapInterface;
+use Sylius\Bundle\CoreBundle\Sitemap\Model\SitemapUrlInterface;
+use Sylius\Bundle\CoreBundle\Sitemap\Provider\UrlProviderInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * @author Arkadiusz Krakowiak <arkadiusz.krakowiak@lakion.com>
+ */
+class SitemapBuilderSpec extends ObjectBehavior
+{
+    function let(
+        SitemapFactoryInterface $sitemapFactory,
+        ContainerInterface $container
+    ) {
+        $this->beConstructedWith(
+            $sitemapFactory,
+            array(
+                'resources' => array(
+                    'product' => array(
+                        'provider' => 'sylius.sitemap.provider.product',
+                    ),
+                    'static' => array(
+                        'provider' => 'sylius.sitemap.provider.static',
+                    ),
+                ),
+                'template' => '@SyliusCore/Sitemap/url_set.xml.twig'
+            )
+        );
+
+        $this->setContainer($container);
+    }
+
+    function it_is_initializable()
+    {
+        $this->shouldHaveType('Sylius\Bundle\CoreBundle\Sitemap\Service\SitemapBuilder');
+    }
+
+    function it_implements_sitemap_builder_interface()
+    {
+        $this->shouldImplement('Sylius\Bundle\CoreBundle\Sitemap\Service\SitemapBuilderInterface');
+    }
+
+    function it_is_container_aware()
+    {
+        $this->shouldHaveType('Symfony\Component\DependencyInjection\ContainerAware');
+    }
+
+    function it_builds_sitemap(
+        SitemapFactoryInterface $sitemapFactory,
+        UrlProviderInterface $productUrlProvider,
+        UrlProviderInterface $staticUrlProvider,
+        SitemapInterface $sitemap,
+        ContainerInterface $container,
+        SitemapUrlInterface $bookUrl,
+        SitemapUrlInterface $homePage
```

Comment:
Ok, it's #justcodenazithings, but you could sort alphabetically this parameters ;)